src/com/cloudogu/ces/cesbuildlib/Git.groovy

```diff
@@ -113,6 +143,25 @@ class Git implements Serializable {
         return repositoryName
     }
 
+    /**
+     * Creates a release on Github and fills it with the changes provided
+     */
+    void addGithubRelease(String releaseVersion, String changes){
+        def repositoryName = getRepositoryName()
+        if (credentials) {
+            script.withCredentials([script.usernamePassword(credentialsId: credentials, usernameVariable: 'GIT_AUTH_USR', passwordVariable: 'GIT_AUTH_PSW')]) {
+                def body = "'{\"tag_name\": \"${releaseVersion}\", \"target_commitish\": \"master\", \"name\": \"${releaseVersion}\", \"body\":\"${changes}\"}'"
+                def apiUrl = "https://api.github.com/repos/${repositoryName}/releases"
+                def flags = "--request POST --data ${body} --header \"Content-Type: application/json\""
+                def username='\$GIT_AUTH_USR'
+                def password='\$GIT_AUTH_PSW'
```

Comment:
What is this for?

Reply:
The only way I found for Jenkins to acknowledge the credentials. Tried a more direct way, but that failed: https://github.com/cloudogu/ces-build-lib/pull/32/commits/4d7de37f1564f003ec084ae31465ef67561054cc

Reply:
I think you could just use
`"curl -u ${GIT_AUTH_USR}:${GIT_AUTH_PSW} ${flags} ${apiUrl}"`
which will lead to groovy doing the replacement. Maybe you need to use `${env.GIT_AUTH_USR}`.

Either way - are you sure the credentials are not shown in the log? Passing password in a plain text shell command is risky.
Unfortunately, `curl` seems to be able to read ENV directly. But what about  [this beautiful solution](https://stackoverflow.com/a/33818945/)?

Reply:
On 2nd thought this might be too complex.

Reply:
I tried this in a Jenkins project.
The console output will be like: `+ curl -u ****:**** ....` so it seems that username and password are hidden.

Reply:
When i change to the `${env.GIT_AUTH_USR}`-solution, it won't work in jenkins anymore. It says `Scripts not permitted to use method groovy.lang.GroovyObject getProperty java.lang.String`.

Reply:
And i think the current way how it is done is more readable than the way how it's suggested on the stackoverflow post.
So is it ok as it ist or should it be changed anyway?